qa/rpc-tests/listsinceblock.py

```diff
@@ -76,5 +76,93 @@ def run_test (self):
                 break
         assert_equal(found, True)
 
+    def test_double_spend(self):
+        '''
+        This tests the case where the same UTXO is spent twice on two separate
+        blocks as part of a reorg.
+
+             ab0
+          /       \
+        aa1 [tx0]   bb1 [tx1]
```

Comment:
Maybe change tx0 and tx1 to tx1 and tx2 to match code. Also code is generating 6 and 7 blocks instead of 3 and 4, so maybe that could be changed to match the example.